crates/compiler/test_gen/src/gen_num.rs

```diff
@@ -1484,6 +1484,172 @@ fn gte_f64() {
+    assert_evals_to!("Num.isApproxEq { value: -0.0, referenceValue: -0.0 }",true,bool);
+    assert_evals_to!("Num.isApproxEq { value: 0.00000001, referenceValue: 0.0 }",false,bool);
+    assert_evals_to!("Num.isApproxEq { value: 0.0, referenceValue: 0.00000001 }",false,bool);
+    assert_evals_to!("Num.isApproxEq { value: -0.00000001, referenceValue: 0.0 }",false,bool);
+    assert_evals_to!("Num.isApproxEq { value: 0.0, referenceValue: -0.00000001 }",false,bool);
+    assert_evals_to!("Num.isApproxEq { value: 0.0, referenceValue: 1e-40 }",true,bool);
+    assert_evals_to!("Num.isApproxEq { value: 1e-40, referenceValue: 0.0 }",true,bool);
+    assert_evals_to!("Num.isApproxEq { value: 1e-40, referenceValue: 0.0 }",false,bool);
+    assert_evals_to!("Num.isApproxEq { value: 0.0, referenceValue: 1e-40 }",false,bool);
+    assert_evals_to!("Num.isApproxEq { value: 0.0, referenceValue: -1e-40 }",true,bool);
+    assert_evals_to!("Num.isApproxEq { value: -1e-40, referenceValue: 0.0 }",true,bool);
+    assert_evals_to!("Num.isApproxEq { value: -1e-40, referenceValue: 0.0 }",false,bool);
+    assert_evals_to!("Num.isApproxEq { value: 0.0, referenceValue: -1e-40 }",false,bool);
+}
+
+#[test]
+#[cfg(any(feature = "gen-llvm", feature = "gen-wasm", feature = "gen-dev"))]
+fn gen_is_approx_eq_min_max() {
+    // Tests involving extreme values (overflow potential)
+    assert_evals_to!("Num.isApproxEq { value: Num.maxF64, referenceValue: Num.maxF64 }",true,bool);
+    assert_evals_to!("Num.isApproxEq { value: Num.maxF64, referenceValue: Num.minF64 }",false,bool);
+    assert_evals_to!("Num.isApproxEq { value: Num.minF64, referenceValue: Num.maxF64 }",false,bool);
+    assert_evals_to!("Num.isApproxEq { value: Num.maxF64, referenceValue: Num.maxF64 / 2 }",false,bool);
+    assert_evals_to!("Num.isApproxEq { value: Num.maxF64, referenceValue: Num.minF64 / 2 }",false,bool);
+    assert_evals_to!("Num.isApproxEq { value: Num.minF64, referenceValue: Num.maxF64 / 2 }",false,bool);
+}
+
+#[test]
+#[cfg(any(feature = "gen-llvm", feature = "gen-wasm", feature = "gen-dev"))]
+fn gen_is_approx_eq_infin() {
+    // Comparison involving infinities
+    assert_evals_to!("Num.isApproxEq { value: ∞, referenceValue: ∞ }",true,bool);
+    assert_evals_to!("Num.isApproxEq { value: -∞, referenceValue: -∞ }",true,bool);
+    assert_evals_to!("Num.isApproxEq { value: -∞, referenceValue: ∞ }",false,bool);
+    assert_evals_to!("Num.isApproxEq { value: ∞, referenceValue: Num.maxF64 }",false,bool);
+    assert_evals_to!("Num.isApproxEq { value: -∞, referenceValue: Num.minF64 }",false,bool);
+}
+
+#[test]
+#[cfg(any(feature = "gen-llvm", feature = "gen-wasm", feature = "gen-dev"))]
+fn gen_is_approx_eq_nan() {
+    // Comparison involving Nan
+    assert_evals_to!("Num.isApproxEq { value: 0.0 / 0.0, referenceValue: 0.0 / 0.0 }",false,bool);
+    assert_evals_to!("Num.isApproxEq { value: 0.0 / 0.0, referenceValue: 0.0 }",false,bool);
```

Comment:
I think this will fail because the default fraction type is `Dec`, not a floating-point one. So `0.0 / 0.0` will default to decimal division, which crashes instead of giving NaN (since `Dec` doesn't have a concept of infinity or NaN).

Instead you can do `0f64 / 0` or `0f32 / 0` to get NaN here, depending on whether you want to test 32-bit or 64-bit floats. (Probably good to test both!)